docs/feature-flags.md

```diff
@@ -0,0 +1,116 @@
+# 🚦 Feature Flagging — Tech Docs
+
+This document explains how our **client‑side** feature‐flag system works, how flags are stored, and how to add or consume a flag in daily development.
+
+---
+
+## 1. What lives where?
+
+```
+src/
+  feature-flags/
+    flags.ts     # canonical list, default states, metadata
+    store.ts     # runtime state resolver (localStorage + URL)
+    panel.tsx    # React component to disable and enable a feature flag
+    hooks.ts     # React hooks and component to disable code
+```
+
+*There is **no** backend call. Everything happens in the browser.*
+
+---
+
+## 2. Persistence & update flow
+
+| Step | Source           | Example                              | Notes                                        |
+| ---- | ---------------- | ------------------------------------ | -------------------------------------------- |
+| 1    | **Default**      | `defaultEnabled` in `flags.ts`       | Baseline value shipped in the bundle.        |
+| 2    | **localStorage** | `__ff_overrides__ => { "foo": true }` | Persists your choices across reloads.        |
+| 3    | **URL params**   | `?ff=foo,bar` or `?ff_foo=false`    | Highest priority; great for shareable links. |
+
+**Setting a flag with our API does two things:**
+
+1. Writes to localStorage so the change sticks.
```

Comment:
I was considering the useContext + localstorage. but url + localstorage is also good choice.